pkg/server/plugin/bundlepublisher/awss3/client.go

```diff
@@ -25,9 +25,20 @@ func newAWSConfig(ctx context.Context, c *Config) (aws.Config, error) {
 		cfg.Credentials = credentials.NewStaticCredentialsProvider(c.AccessKeyID, c.SecretAccessKey, "")
 	}
 
+	if c.Endpoint != "" {
+		cfg.BaseEndpoint = aws.String(c.Endpoint)
+	}
+
 	return cfg, nil
 }
 
 func newS3Client(c aws.Config) (simpleStorageService, error) {
-	return s3.NewFromConfig(c), nil
+	options := func(options *s3.Options) {}
```

Comment:
Prefer the use of `var` for zero value.
```suggestion
	var options func(options *s3.Options)
```